### DISCUSSION TITLE
Nuxt Vision for 2023

### DISCUSSION THREAD

**Discussion by danielroe** [nuxt/nuxt#15979] [Announcements]
This past year has been an exciting one. Looking into the new year, there is a lot we have planned as a team and we'd love to share it with you. :sparkles:

- ✨ [Full article](https://nuxt.com/vision-2023)
- 🐦 [Tweet](https://twitter.com/nuxt_js/status/1615436763933323264)

**Comment by maximedaraize** [comment]
Concerning the [The New Release Cycle](https://nuxt.com/vision-2023#a-new-release-cycle) mention in the article. Will there be a **project** inside the Nuxt repo the follow the progress of those releases?

**Comment by vanling** [comment]
👀 No pressure 😉  but looking forward to the Nuxt vision 2024 👀  Been an amazing year for Nuxt <3

**Comment by kaikaibenkai** [comment]
2024 is coming ✨ Thanks Nuxt Team, Nuxt 3 is great!

**Comment by martinszeltins** [comment]
Looking forward to the 2024 vision also 🚀

**Comment by jackmahoney** [comment]
Please **do not** release v4. The v2-v3 break was too traumatic for some organisations and many devs I know lost faith in Nuxt or jumped to React because of it. My business depends entirely on Nuxt2 and the project to upgrade to v3 has caused a lot of extra work and cost for no noticeable benefits for end users. 
- All our nuxt2 page routing and middleware was suddenly unsupported
- All our nuxt2 UI libraries are abandoned 
- Our Vue2/webpack plugins and now worthless

Please be a reliable, dependable choice for small teams and maintain long term v3. You should **not** be releasing breaking changes every year! The plugin and ecosystem churn is absolutely brutal and will result in more developers abandoning Nuxt. You seem to be favouring new developer experience over your core users that have been here for years and now suffer under your changes that seem to be made on a whim.

For instance:
- Why did you need to change the way dynamic routes are handled in v3? `_id` -> `[id]`. Why was it worth creating a breaking change for this?

I can't finish this because it makes me too depressed thinking how much longer we still have to go on v3 migration and knowing that all that time is wasted as it doesn't deliver anything to our customers. And then I learn that v4 will be coming soon... you guys have to be joking.

**Reply by martinszeltins** [reply]
@jackmahoney Nuxt 4.0 won't have any new features, only small, careful breaking changes with smooth upgrade _(from what I heard @danielroe say)_. I don't think it is fair to never have any breaking changes in a project, that is not realistic or good for the project's future. How will we ever innovate without any breaking changes?

**Reply by jackmahoney** [reply]
@martinszeltins thanks for your considerate response. You are right that breaking changes are occasionally needed but I believe the rate of change needs to be balanced with the needs of the library consumers. I'm worries about another gruelling upgrade path and wanted to raise that. I believe a strongly defined API that does not change is vital for the success of a framework. Consistency allows a strong ecosystem to grow. Many plugins and libraries were left abandoned after v3 changes. I don't want to see that happen again with v4. See what spring boot has done with their release cadence. Major versions are a very rare occurrence and one never has an issue with dependencies or deprecations.

**Reply by TheAlexLichter** [reply]
My response in https://github.com/nuxt/nuxt/issues/26780

**Reply by jackmahoney** [reply]
Thank you

**Reply by kasadawa** [reply]
@jackmahoney totally agree with you. We have a huge project, which was written in nuxt 2, we just moved to nuxt 3, while trying to make it stable, we saw nuxt 4 is on the road. @danielroe you don't realise how this looks in the eyes of developers and stakeholders, but it seems like a waste. There are still a lot of projects, which are on nuxt 2. My personal feeling is that there is no direction, no vision, no proper measures and proper technical decisions. I understand the idea behind the changes, but those things could be seen beforehand and the proper decision could be made. It feels like going back and forth. And the propaganda that this is the proper way to go will always be there.

If you think that the changes aren't that big and the effort is not big to migrate to v4, then why dont you make a CLI based tooling for migrating it ?

**Reply by danielroe** [reply]
@kasadawa In many cases no changes in end user  projects will be needed at all. And yes, the codemod team have kindly produced CLI based migrations for the situations where people need it.

We have not yet released v4, but user migration existence is absolutely top of my mind.

If you try `compatibilityVersion: 4`, even at this stage, I think you will be pleasantly surprised.
